### ISSUE TITLE
Target process crashes on detach/script reload with Java hooks installed (9.1.6)

### ISSUE THREAD

**Issue by hedger** [frida/frida-java#18]
`frida -U -l java.js com.android.terminal`
On editing script or detaching from target process it crashes.
Crashes don't happen if I don't set up a hook.

java.js:
```
function onLoad() {
    console.log("Loaded.")

    Java.perform( function() { 
      URL = Java.use('java.net.URL');
      URL.getHost.implementation = function () {
        console.log("inside hook!");
        return this.getHost();
      };
    });

    Java.perform( function() { 
      url = URL.$new('http://www.ikke.no/');
      host = url.getHost();
      console.log('host=' + host);
    });
}
onLoad();
```
Android 6.0.1, CM13. Device is Wileyfox Swift.

**Comment by hedger** [comment]
Any updates on this issue?

**Comment by oleavr** [comment]
Just released 9.1.7 with Android N support. Will investigate this next.

**Comment by oleavr** [comment]
Fixed in latest versions.
